### PR TITLE
completed task 7

### PR DESCRIPTION
**Pull request by Dominic20227**

### CODE REVIEW

**Review by popatre — CHANGES_REQUESTED** [review]
- UI updates and lets us know that the votes have been incremented - nice one 👍
- Nice to see the votes are persistent between refreshes

- [ ] We need to provide some feedback that something has gone wrong i.e. if the Internet connection is turned off, or the server has failed to do the request, we need to let the user know. At the moment, it just says please wait.
At the moment, there's no feedback or the votes seem to increment, but actually the request has failed.
- [ ] You should ideally handle the updating of your votes using optimistic rendering.At the moment, if im on a slow internet connection, and click the vote button, theres a delay between clicking and something happening. We get your loading message for ages, then the votes - its not great UX.
For a better UX - use optimistic rendering - 

https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#optimistic-rendering

- [ ] There also a really annoying flashing when you vote due to the loading screen thing

File: src/components/ArticleDetail.jsx [hunk 1]
```diff
@@ -16,8 +17,22 @@ function ArticleDetail({ singleArticle }) {
...
+  function handleUpvote(articleID) {
+    setSingleArticle({ ...singleArticle, votes: singleArticle.votes + 1 });
+
+    upVoteArticle(articleID)
+      .then((updatedArticle) => {
+        setSingleArticle(updatedArticle);
+      })
+      .catch((err) => {
+        if (err) {
+          setErr("Something went wrong");
```
Comment (ghost) [inline src/components/ArticleDetail.jsx:29]: I like that this is there, to the user the number goes up then they see the message. Is there a way to perhaps either reset the number when this happens, or avoid it going up?

**Review by ghost — APPROVED** [review]
This works well and persists on a refresh. Nice optimistic rendering as well. Consider the error handling. It works really well, but does show an updated vote before the error message giving a bit of mixed messaging.

**Review by popatre — CHANGES_REQUESTED** [review]
Nice message for the failed patch.

However, theres confusing UX now where the seconds/dates change on the comments when you upvote. I presume these should remain as they originally were, as they are when they posted it, but instead, all the times now change everytime you upvote

**Comment by ghost** [comment]
Much better :D nice job, merge and move on.

**Pull request merged**